poutyne/framework/callbacks/gradient_tracking.py

```diff
@@ -0,0 +1,84 @@
+import torch
+from torch import tensor
+
+from poutyne.framework import Callback
+
+
+class GradientTracker(Callback):
+    def __init__(self, writer):
+        super().__init__()
+        self.writer = writer
+        self.layers_name = []
+        self.number_layers = 0
+
+        self.running_mean = None
+        self.running_variance = None
+        self.running_m2 = None  # use to calculate the variance
+        self.running_min = None
+        self.running_max = None
+
+    def on_train_begin(self, logs):
+        for layer_name, layer_gradient in self.model.model.named_parameters():
+            if layer_gradient.requires_grad and ("bias" not in layer_name):
+                self.layers_name.append(layer_name)
+
+        self.number_layers = len(self.layers_name)
+
+    def on_epoch_begin(self, epoch, logs):
+        self.running_mean = torch.zeros([self.number_layers])
+        self.running_variance = torch.zeros([self.number_layers])
+        self.running_m2 = torch.zeros([self.number_layers])
+        self.running_min = torch.zeros([self.number_layers])
+        self.running_max = torch.zeros([self.number_layers])
+
+    def on_batch_end(self, batch, logs):
+        self._on_batch_end_write(batch)
+
+    def _on_batch_end_write(self, batch):
+        batch_layer_means = []
+        batch_layer_min = []
+        batch_layer_max = []
+        for layer_name, layer_gradient in self.model.model.named_parameters():
+            if layer_gradient.requires_grad and ("bias" not in layer_name):
+                batch_layer_means.append(layer_gradient.grad.abs().mean())
+                batch_layer_min.append(layer_gradient.grad.abs().min())
+                batch_layer_max.append(layer_gradient.grad.abs().max())
```

Comment:
Optimization : Very small but storing layer_params.grad.abs() in a variable instead of always calling it would be faster.

Reply:
Good call.

Reply:
6856b8d